### PR TITLE
Allow optional map key in schema of parquet files

### PR DESCRIPTION
**Pull request by taiyang-li**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
When HDFS Parquet files are written by flink, and there is a map type column in the schema, then its key type is default optional. It will cause CH to throw exception when reading such parquet files because arrow lib forbidden optional key in map.

Related pr: https://github.com/ClickHouse/arrow/pull/17

### CODE REVIEW

**Comment by taiyang-li** [comment]
@Avogar  Can you also review this submodule change? Thanks !

**Comment by Avogar** [comment]
Can you write a test for it? For example, generate a file with optional map key and try to read it.

**Comment by taiyang-li** [comment]
> Can you write a test for it? For example, generate a file with optional map key and try to read it.

 Sorry, I tried before but it is impossible to generate parquet file with map type which contains optional key using arrow lib. The only possible way is Flink.

**Comment by Avogar** [comment]
> The only possible way is Flink.
Can you use it to generate a small file for tests?

**Comment by taiyang-li** [comment]
[example.zip](https://github.com/ClickHouse/ClickHouse/files/11144583/example.zip) 


$ hadoop jar parquet-tools-1.9.0.jar  schema /user/liyang/testdata/example.parquet  
message flink_schema {
  optional int64 id;
  optional group data (MAP) {
    repeated group key_value (MAP_KEY_VALUE) {
      optional binary key (UTF8);
      optional binary value (UTF8);
    }
  }
} 


@Avogar above is an example parquet file with optional map key.

File: tests/queries/0_stateless/02703_optional_map_key_in_parquet.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
```
Comment (Avogar) [inline tests/queries/0_stateless/02703_optional_map_key_in_parquet.sh:2]:
```suggestion
# Tags: no-fasttest

```

**Comment by taiyang-li** [comment]
failed tests seem not related to this pr.

**Comment by Avogar** [comment]
`Stateless tests (debug) [4/5] — fail: 1` is related. The new test is failed

**Comment by taiyang-li** [comment]
@Avogar Sorry, I have no idea how to fix this failed tests in debug mode. It seems impossible to fix it under framework of clickhouse because key type of map in CH should always be non-nullable.

**Pull request closed without merging**